torch/csrc/autograd/python_variable.cpp

```diff
@@ -292,10 +294,9 @@ int THPVariable_set_backwards_hooks(THPVariable *self, PyObject *obj)
   Py_XINCREF(obj);
   Py_XDECREF(self->backward_hooks);
   self->backward_hooks = obj;
+  self->cdata->pre_hooks.clear();
   if (obj) {
-    self->cdata->backward_hook.reset(new PyGradHook(obj));
-  } else {
-    self->cdata->backward_hook.reset();
+    self->cdata->pre_hooks.emplace_back(new PyFunctionPreHook(obj, 0));
   }
```

Comment:
it will remove them, but only from the Variable. The content of the field from the Variable doesn't matter, the important field is that vector inside the creator that will still remember the old value.